src/components/tasks/card/TaskStatusEditMode.tsx

```diff
@@ -1,42 +1,76 @@
 import { BACKEND_TASK_STATUS } from '@/constants/task-status';
-import task from '@/interfaces/task.type';
+import task, { CardTaskDetails } from '@/interfaces/task.type';
+import { useState } from 'react';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import { SmallSpinner } from './SmallSpinner';
+import { SavedCheckmark } from './SavedCheckmark';
+import classNames from '@/components/tasks/card/card.module.scss';
+import { PENDING, SAVED } from '../constants';
+import { useUpdateTaskMutation } from '@/app/services/tasksApi';
 
 type Props = {
     task: task;
-    updateTask: (changeId: string, changeObject: { status: string }) => void;
+    setEditedTaskDetails: React.Dispatch<React.SetStateAction<CardTaskDetails>>;
 };
 
 // TODO: remove this after fixing the card beautify status
 const beautifyStatus = (status: string) => status.split('_').join(' ');
 const taskStatus = Object.entries(BACKEND_TASK_STATUS);
 
-const TaskStatusEditMode = ({ task, updateTask }: Props) => {
+const TaskStatusEditMode = ({ task, setEditedTaskDetails }: Props) => {
+    const { ERROR } = ToastTypes;
+    const [saveStatus, setSaveStatus] = useState('');
+    const [updateTask] = useUpdateTaskMutation();
+
     const onChangeUpdateTaskStatus = ({
```

Comment:
onChangeUpdateTaskStatus function. The function is supposed to update the task status and set the saveStatus variable to SAVED if the update is successful. However, the function is not checking the response from the updateTask mutation. If the mutation fails, the saveStatus variable will not be set to SAVED and the user will not be notified of the error.

Reply:
If the mutation fails, it will go inside catch block and will show the toast to the user with the error message. Function is not checking the response from the updateTask mutation inside “then” block because response will be null if mutation is successful.
Please correct me, if my understanding is wrong.